File: server/src/main/java/by/brawl/ws/spell/SpellLogic.java
package by.brawl.ws.spell;

import by.brawl.ws.holder.BattlefieldHolder;

import java.util.Set;

/**
 * Default class description.
 *
 * @author P.Sinitsky.
 *         Created on 14.04.2017.
 */
public interface SpellLogic {

    BattlefieldHolder cast(BattlefieldHolder battlefieldHolder,
                           String senderId,
                           Integer victimPosition,
                           Boolean forEnemy);
    String getId();
    Boolean getTargetable();
    Set<Integer> getMyTargets();
    Set<Integer> getEnemyTargets();
}

File: server/src/main/java/by/brawl/ws/spell/SpellLogic.kt
package by.brawl.ws.spell

import by.brawl.ws.holder.BattlefieldHolder

interface SpellLogic {

    fun cast(battlefieldHolder: BattlefieldHolder,
             senderId: String,
             victimPosition: Int?,
             forEnemy: Boolean?): BattlefieldHolder

    val id: String
    val targetable: Boolean
    val myTargets: Set<Int>
    val enemyTargets: Set<Int>
}